transports/quic/tests/smoke.rs

```diff
@@ -0,0 +1,362 @@
+        e => panic!("{:?}", e),
+    };
+
+    match b.next().await {
+        Some(SwarmEvent::ConnectionEstablished { .. }) => {}
+        e => panic!("{:?}", e),
+    };
+
+    match a.next().await {
+        Some(SwarmEvent::ConnectionEstablished { .. }) => {}
+        e => panic!("{:?}", e),
+    };
+
+    assert!(b.next().now_or_never().is_none());
+
+    match a.next().await {
+        Some(SwarmEvent::Behaviour(RequestResponseEvent::Message {
+            message:
+                RequestResponseMessage::Request {
+                    request: Ping(ping),
+                    channel,
+                    ..
+                },
+            ..
+        })) => {
+            a.behaviour_mut()
+                .send_response(channel, Pong(ping))
+                .unwrap();
+        }
+        e => panic!("{:?}", e),
+    }
+
+    match a.next().await {
+        Some(SwarmEvent::Behaviour(RequestResponseEvent::ResponseSent { .. })) => {}
+        e => panic!("{:?}", e),
+    }
+
+    match b.next().await {
+        Some(SwarmEvent::Behaviour(RequestResponseEvent::Message {
+            message:
+                RequestResponseMessage::Response {
+                    response: Pong(pong),
+                    ..
+                },
+            ..
+        })) => assert_eq!(data, pong),
+        e => panic!("{:?}", e),
+    }
+
+    a.behaviour_mut().send_request(
+        &Swarm::local_peer_id(&b),
+        Ping(b"another substream".to_vec()),
+    );
+
+    assert!(a.next().now_or_never().is_none());
+
+    match b.next().await {
+        Some(SwarmEvent::Behaviour(RequestResponseEvent::Message {
+            message:
+                RequestResponseMessage::Request {
+                    request: Ping(data),
+                    channel,
+                    ..
+                },
+            ..
+        })) => {
+            b.behaviour_mut()
+                .send_response(channel, Pong(data))
+                .unwrap();
+        }
+        e => panic!("{:?}", e),
+    }
+
+    match b.next().await {
+        Some(SwarmEvent::Behaviour(RequestResponseEvent::ResponseSent { .. })) => {}
+        e => panic!("{:?}", e),
+    }
+
+    match a.next().await {
+        Some(SwarmEvent::Behaviour(RequestResponseEvent::Message {
+            message:
+                RequestResponseMessage::Response {
+                    response: Pong(data),
+                    ..
+                },
+            ..
+        })) => assert_eq!(data, b"another substream".to_vec()),
+        e => panic!("{:?}", e),
+    }
+
+    Ok(())
+}
+
+#[derive(Debug, Clone)]
+struct PingProtocol();
+
+#[derive(Clone)]
+struct PingCodec();
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct Ping(Vec<u8>);
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct Pong(Vec<u8>);
+
+impl ProtocolName for PingProtocol {
+    fn protocol_name(&self) -> &[u8] {
+        "/ping/1".as_bytes()
+    }
+}
+
+#[async_trait]
+impl RequestResponseCodec for PingCodec {
```

Comment:
Is there any reason not to reuse the existing `libp2p::ping`?

Reply:
for test purposes it's useful to be able to add large payloads otherwise the stream will only contain one packet